src/inmanta/logging.py

```diff
@@ -99,15 +117,103 @@ def __init__(self, stream: TextIO = sys.stdout) -> None:
         self.set_log_formatter(formatter)
 
+        self._logger_mode = LoggerMode.OTHER
+        self._default_log_level_factory = logging.getLogRecordFactory()
+        logging.setLogRecordFactory(self.custom_log_record_factory)
+        self._source_file_to_module_cache: dict[str, Optional[str]] = {}
+        # A cache for the `_get_module_name_for_source_file()` method.
+
         logging.root.handlers = []
         logging.root.addHandler(self._handler)
         logging.root.setLevel(0)
 
+    def _get_module_name_for_source_file(self, path_source_file: str) -> Optional[str]:
+        """
+        Return the module name that the given `path_source_file` belongs to or None if `path_source_file`
+        doesn't belong to a module.
+        """
+        if path_source_file in self._source_file_to_module_cache:
+            return self._source_file_to_module_cache[path_source_file]
+        dirs_containing_modules = []
+        if Project._project:
+            # Directories containing V1 modules
+            dirs_containing_modules += [p for p in Project._project._metadata.modulepath]
+        if (
+            "inmanta_plugins" in sys.modules
+            and sys.modules["inmanta_plugins"].__spec__
+            and sys.modules["inmanta_plugins"].__spec__.submodule_search_locations
+        ):
+            # Directories containing v2 modules
+            dirs_containing_modules += [str(s) for s in sys.modules["inmanta_plugins"].__spec__.submodule_search_locations]
+        result = None
+        for mod_dir in dirs_containing_modules:
+            if path_source_file.startswith(mod_dir):
+                rel_path = path_source_file[len(mod_dir) :].strip("/")
+                if rel_path:
+                    result = rel_path.split("/", maxsplit=1)[0]
+
+        self._source_file_to_module_cache[path_source_file] = result
+        return result
+
+    def custom_log_record_factory(
+        self,
+        name: str,
+        level: int,
+        pathname: str,
+        lineno: int,
+        msg: object,
+        args: Union[tuple[...] | dict[str, object]],
+        exc_info: Optional[tuple[type[BaseException], BaseException, types.TracebackType]],
+        func: Optional[str] = None,
+        sinfo: Optional[str] = None,
+        **kwargs: object,
+    ) -> logging.LogRecord:
+        """
+        This log record factory makes sure that the name of the log record is updated
+        in the following way while executing in the "compiler" or "exporter" logger mode:
+
+        * The name of the Inmanta module: When the log record was created by a source file in an Inmanta module.
+        * compiler: When executing in compiler mode and the log record doesn't come from an Inmanta module.
+        * exporter: When executing in exporter mode and the log record doesn't come from an Inmanta module.
+        """
+        new_logger_name: str
+        if self._logger_mode is LoggerMode.COMPILER or self._logger_mode is LoggerMode.EXPORTER:
```

Comment:
```suggestion
        if self._logger_mode in [LoggerMode.COMPILER, LoggerMode.EXPORTER]:
```
I just like it more but feel free to ignore

Reply:
Indeed. Much better.